x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -124,7 +124,12 @@ public void run() {
         client.admin().indices().getIndex(getIndexRequest, new ActionListener<>() {
             @Override
             public void onResponse(GetIndexResponse getIndexResponse) {
-                validateMappings(getIndexResponse);
+                try {
+                    validateMappings(getIndexResponse);
```

Comment:
Obvious spot where we have a non-`wrap` listener and this would just bubble up all the way if the request is executed locally on `SAME`.